pac4j-couch/src/main/java/org/pac4j/couch/profile/service/CouchProfileService.java

```diff
@@ -0,0 +1,181 @@
+package org.pac4j.couch.profile.service;
+
+import org.pac4j.couch.profile.CouchProfile;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.ektorp.CouchDbConnector;
+import org.ektorp.DocumentNotFoundException;
+import org.ektorp.ViewQuery;
+import org.ektorp.ViewResult;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.password.PasswordEncoder;
+import org.pac4j.core.profile.definition.CommonProfileDefinition;
+import org.pac4j.core.profile.service.AbstractProfileService;
+import org.pac4j.core.util.CommonHelper;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The MongoDB profile service.
+ *
+ * @author Elie Roux
+ * @since 2.0.0
+ */
+public class CouchProfileService extends AbstractProfileService<CouchProfile> {
+
+    private CouchDbConnector couchDbConnector;
+    
+    public static final String COUCH_ID = "_id";
+    public static ObjectMapper objectMapper = new ObjectMapper();
+
+    public CouchProfileService() {
+    	setIdAttribute(COUCH_ID);
+    }
+
+    public CouchProfileService(final CouchDbConnector couchDbConnector) {
+    	setIdAttribute(COUCH_ID);
+        this.couchDbConnector = couchDbConnector;
+    }
+
+    public CouchProfileService(final CouchDbConnector couchDbConnector, final String attributes) {
+    	setIdAttribute(COUCH_ID);
+        this.couchDbConnector = couchDbConnector;
+        setAttributes(attributes);
+    }
+
+    public CouchProfileService(final CouchDbConnector couchDbConnector, final String attributes, final PasswordEncoder passwordEncoder) {
```

Comment:
Could you make all the other constructor defer to this one using this(...) in order not to repeat the construction logic?